app/helpers/gtl_helper.rb

```diff
@@ -0,0 +1,125 @@
+module GtlHelper
+  def gtl_selected_records
+    records = params.try(:[], :rec_ids) || @edit.try(:[], :pol_items) || @edit.try(:[], :object_ids)
+
+    if records.present?
+      records = records.map(&:to_i) if records.first.kind_of?(String)
+      records = records.map(&:id) unless records.first.kind_of?(Integer)
+    end
+
+    records
+  end
+
+  def miq_tile_view
+    content_tag(
+      'miq-tile-view', '',
+      "ng-if"            => "dataCtrl.gtlType === 'grid' || dataCtrl.gtlType === 'tile'",
+      "ng-class"         => "{'no-action': dataCtrl.initObject.showUrl === ''}",
+      "settings"         => "dataCtrl.settings",
+      "per-page"         => "dataCtrl.perPage",
+      "rows"             => "dataCtrl.gtlData.rows",
+      "on-row-click"     => "dataCtrl.onItemClicked(item, event)",
+      "on-sort"          => "dataCtrl.onSort(headerId, isAscending)",
+      "on-item-selected" => "dataCtrl.onItemSelect(item, isSelected)",
+      "load-more-items"  => "dataCtrl.onLoadNext(start, perPage)",
+      "columns"          => "dataCtrl.gtlData.cols",
+      "type"             => "dataCtrl.gtlType === 'grid' ? 'small' : 'big'"
+    )
+  end
+
+  def miq_data_table
+    content_tag(
+      'miq-data-table', '',
+      "ng-if"            => "dataCtrl.gtlType === 'list'",
+      "ng-class"         => "{'no-action': dataCtrl.initObject.showUrl === false}",
+      "settings"         => "dataCtrl.settings",
+      "per-page"         => "dataCtrl.perPage",
+      "rows"             => "dataCtrl.gtlData.rows",
+      "on-row-click"     => "dataCtrl.onItemClicked(item, event)",
+      "on-sort"          => "dataCtrl.onSort(headerId, isAscending)",
+      "load-more-items"  => "dataCtrl.onLoadNext(start, perPage)",
+      "on-item-selected" => "dataCtrl.onItemSelect(item, isSelected)",
+      "columns"          => "dataCtrl.gtlData.cols"
+    )
+  end
+
+  # This method collects all the data comming from side channels
+  # so that `render_gtl` can be a pure function.
+  #
+  def render_gtl_outer(no_flash_div)
+    parent_id = @report_data_additional_options.try(:[], :parent_id)
+
+    options = {
+      :model_name                     => model_to_report_data,
+      :no_flash_div                   => no_flash_div || false,
+      :gtl_type_string                => @gtl_type,
+      :active_tree                    => (x_active_tree unless params[:display] || @use_action),
```

Comment:
I find this very hard to read, imho it would be easier to read something like:
```ruby
(!params[:display] && x_active_tree) || @use_action
```

Reply:
@ohadlevy I don't think what you wrote is equivalent to the pice of code that you're commenting.

Reply:
It's not. The first returns x_active_tree or null the latter returns x_active_tree or @use_action. The change could actually cause a regression if @use_action was not bool and for example some random string was passed as the active_tree. Thanks for making a point on not doing more refactoring in this PR :-D